Lib/javascript/v8/javascriptinit.swg

```diff
@@ -69,11 +69,17 @@ SWIG_V8_GetModule(void *) {
 extern "C"
 #if (NODE_MODULE_VERSION < 0x000C)
 void SWIGV8_INIT (SWIGV8_OBJECT exports)
+#elif (NODE_MODULE_VERSION < 0x0040)
+void SWIGV8_INIT (SWIGV8_OBJECT exports, SWIGV8_VALUE /*module*/, void*)
 #else
-void SWIGV8_INIT (SWIGV8_OBJECT exports, SWIGV8_OBJECT /*module*/)
+void SWIGV8_INIT (SWIGV8_OBJECT exports, SWIGV8_VALUE /*module*/, v8::Local<v8::Context> context, void*)
 #endif
 {
-  SWIG_InitializeModule(static_cast<void *>(&exports));
+#if NODE_MODULE_VERSION < 0x0040
+  v8::Local<v8::Context> context = SWIGV8_CURRENT_CONTEXT();
+#endif
+
+  SWIG_InitializeModule(static_cast<void *>(&context));
```

Comment:
Just in case. @chenhengqi, I guess your remark was triggered by this `&context`. Well, it doesn't really matter from efficiency viewpoint if we keep passing this reference, or dereference it at cast-from-void-* point and pass v8::Local by value. It's more of a readability issue. Rationale was that passing `v8::Local<T>` by value is better aligned with the spirit of v8...

Reply:
Given Local<> is just holds a pointer, the performance isn't an issue. Also v8.h mentions Local should be passed by value: https://github.com/v8/v8/blob/d9f3d4290e8dcbec9a82e4d6d07bd7e9528ae8e4/include/v8.h#L175-L178.

Reply:
I wish it was legitimate to cast `context` to `void *` **here**. Of course one can probably make adjustments so that SWIG_InitializeModule would accept `Local<Context>`. Should I look into it?

Reply:
Yeah why not if you can.

Reply:
Note that I wrote "probably." It's not like it's impossible (nothing is), but it won't be few additional lines. SWIG_InitializeModule is common for all languages, and so are structures it uses. Taking https://github.com/swig/swig/blob/master/Lib/swiginit.swg#L121 for example. Using `Local<Context>` as type for `clientdata` would trigger V8-specific way to check if it's initialized. In other words it would be more #ifdef-s, which might be an overkill. After all, it's just **one** language among many...

It's late here, I'm going offline for today, and I have meetings tomorrow... So think about it, and give your verdict, today or tomorrow...

Cheers.

Reply:
Leave it as is then. Thanks.

Reply:
@dot-asm Thanks for the details.

Reply:
Hmmm... You know what? That reference to L121 means that this line we are commenting on is wrong. In sense that it would be inappropriate to save reference to local variable (if it is to be dereferenced later on). As mentioned, I'm a bit busy today, but I should be able to provide more adequate solution during the day...

Reply:
It looks like it was assessment double-fault:-) Well, it was late after all. First of all, L121 has nothing to do with what we're talking about. And second faulty assessment was that it would be too disruptive for other languages. I've got caught thinking in multi-module terms, while SWIG is single-module in spirit. (Just in case, "multi-module" means separated headers and possible multiple compile units.) Either way `&context` is gone, so check it out. There also some additional minor cleanups...